src/main/java/net/sf/jabref/importer/fileformat/MedlineImporter.java

```diff
@@ -84,50 +140,532 @@ public boolean isRecognizedFormat(BufferedReader reader) throws IOException {
+            }
+            if (bookDocument.getBook() != null) {
+                addBookInformation(fields, bookDocument.getBook());
+            }
+        }
+
+        if (currentArticle.getPubmedBookData() != null) {
+            PubmedBookData bookData = currentArticle.getPubmedBookData();
+            putIfNotNull(fields, "pubstatus", bookData.getPublicationStatus());
+        }
+
+        BibEntry entry = new BibEntry(IdGenerator.next(), "article"); // id assumes an existing database so don't create one here
+        entry.setField(fields);
+
+        bibItems.add(entry);
+    }
+
+    private void addBookInformation(HashMap<String, String> fields, Book book) {
+        if (book.getPublisher() != null) {
+            Publisher publisher = book.getPublisher();
+            putIfNotNull(fields, "publocation", publisher.getPublisherLocation());
+            putStringFromSerializableList(fields, "publisher", publisher.getPublisherName().getContent());
+        }
+        if (book.getBookTitle() != null) {
+            BookTitle title = book.getBookTitle();
+            putStringFromSerializableList(fields, "title", title.getContent());
+        }
+        if (book.getPubDate() != null) {
+            addPubDate(fields, book.getPubDate());
+        }
+        if (book.getAuthorList() != null) {
+            List<AuthorList> authorLists = book.getAuthorList();
+            //authorLists size should be one
+            if (authorLists.size() == 1) {
+                for (AuthorList authorList : authorLists) {
+                    handleAuthors(fields, authorList);
+                }
+            }
+        }
+
+        putIfNotNull(fields, "volume", book.getVolume());
+        putIfNotNull(fields, "edition", book.getEdition());
+        putIfNotNull(fields, "medium", book.getMedium());
+        putIfNotNull(fields, "reportnumber", book.getReportNumber());
+
+        if (book.getELocationID() != null) {
+            for (ELocationID id : book.getELocationID()) {
+                addElocationID(fields, id);
+            }
+        }
+        if (book.getIsbn() != null) {
+            fields.put("isbn", join(book.getIsbn(), ", "));
+        }
+    }
+
+    private void putStringFromSerializableList(HashMap<String, String> fields, String medlineKey,
+            List<Serializable> contentList) {
+        String result = "";
+        for (Serializable content : contentList) {
+            if (content instanceof String) {
+                result += (String) content;
+            }
+        }
+        if (!"".equals(result)) {
+            fields.put(medlineKey, result);
+        }
+    }
+
+    private void addContributionDate(HashMap<String, String> fields, ContributionDate contributionDate) {
+        if ((contributionDate.getDay() != null) && (contributionDate.getMonth() != null)
+                && (contributionDate.getYear() != null)) {
+            String result = contributionDate.getYear() + "/" + contributionDate.getMonth() + "/"
```

Comment:
Since the parts are in separate Strings(?) already I think it is OK to concatenate. If they where ints it might have been a different thing. But I also think that - is a better separator.